sstables/sstables.cc

```diff
@@ -490,17 +490,17 @@ typename disk_set_of_tagged_union<TagType, Members...>::serdes disk_set_of_tagge
 
 template <typename TagType, typename... Members>
 future<>
-parse(sstable_version_types v, random_access_reader& in, disk_set_of_tagged_union<TagType, Members...>& s) {
+parse(schema_ptr schema, sstable_version_types v, random_access_reader& in, disk_set_of_tagged_union<TagType, Members...>& s) {
     using disk_set = disk_set_of_tagged_union<TagType, Members...>;
```

Comment:
You are copying schema_ptr:s like crazy. They're pretty lightweight but still that bloats the code with calls to the schema destructor that will never happen.

Can you capture the schema by reference? Need to make sure the top-level caller to parse holds a steay pointer to the schema.

Reply:
I changed it to const& the only caller takes schema from sstable so it's safe. @nyh I hope you won't mind me not replying to all the move comments - I appreciate the time you took to find all of them but happily Avi's suggestion solves the problem.